Sources/PromotedCore/AnomalyHandling/AnomalyModalViewController.swift

```diff
@@ -0,0 +1,200 @@
+import Foundation
+import UIKit
+
+/** Delegate for interactions with the AnomalyModalVC. */
+protocol AnomalyModalViewControllerDelegate: AnyObject {
+  func anomalyModalVCDidDismiss(
+    _ vc: AnomalyModalViewController,
+    shouldShowAgain: Bool
+  )
+}
+
+/**
+ Presents a modal with detailed error message. Clients should not use this
+ class directly.
+
+ Although this class is intended only for internal use in the Promoted metrics
+ library, it is declared public so that ReactNativeMetrics can show the modal
+ for module initialization issues.
+ */
+public class AnomalyModalViewController: UIViewController {
+
+  weak var delegate: AnomalyModalViewControllerDelegate?
+
+  private let partner: String
+  private let contactInfo: [String]
+  private let anomalyType: AnomalyType
+
+  private var shouldShowAgain: Bool
+
+  init(
+    partner: String,
+    contactInfo: [String],
+    anomalyType: AnomalyType,
+    delegate: AnomalyModalViewControllerDelegate?
+  ) {
+    self.partner = partner
+    self.contactInfo = contactInfo
+    self.anomalyType = anomalyType
+    self.delegate = delegate
+    self.shouldShowAgain = true
+    super.init(nibName: nil, bundle: nil)
+    if #available(iOS 13.0, *) {
+      self.isModalInPresentation = true  // Prevent dragging down to dismiss
+    }
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  public override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    guard let view = self.view else { return }
+    let backdropWidth = view.bounds.width - 40
+    let textWidth = backdropWidth - 40
+    let textLayoutFrame = CGRect(x: 0, y: 0, width: textWidth, height: 0)
+
+    let blurEffect = UIBlurEffect(style: .dark)
+    let backdrop = UIVisualEffectView(effect: blurEffect)
+    backdrop.clipsToBounds = true
+    backdrop.layer.cornerRadius = 20
```

Comment:
Side - Interesting.  It feels like this controller contains more view logic than I'd expect.  Is this normal for iOS?  I think it's fine to do this to avoid another class.  I'm asking to learn iOS conventions.

Reply:
For completeness, there's also SwiftUI, which is a much nicer, declarative UI spec. I really like the syntax and the direction it's headed, but it requires iOS 13, and some of our partners build for iOS 11.

You can show UIKit-based View Controllers (which is what we have here) in SwiftUI, but you can't show SwiftUI in pre-iOS 13 apps. So for compatibility, I chose to do this in a UIKit-based solution.